apps/api/src/app/subscribers/usecases/remove-subscriber/remove-subscriber.usecase.ts

```diff
@@ -1,52 +1,89 @@
+    private topicSubscribersRepository: TopicSubscribersRepository,
+    private subscriberPreferenceRepository: SubscriberPreferenceRepository,
+    private preferenceRepository: PreferencesRepository
   ) {}
 
-  async execute(command: RemoveSubscriberCommand) {
-    try {
-      const { environmentId: _environmentId, organizationId, subscriberId } = command;
-      const subscriber = await this.getSubscriber.execute({
-        environmentId: _environmentId,
-        organizationId,
-        subscriberId,
-      });
-
-      await this.invalidateCache.invalidateByKey({
+  async execute({ environmentId: _environmentId, subscriberId }: RemoveSubscriberCommand) {
+    await Promise.all([
+      this.invalidateCache.invalidateByKey({
         key: buildSubscriberKey({
-          subscriberId: command.subscriberId,
-          _environmentId: command.environmentId,
+          subscriberId,
+          _environmentId,
         }),
-      });
+      }),
+      this.invalidateCache.invalidateQuery({
+        key: buildFeedKey().invalidate({
+          subscriberId,
+          _environmentId,
+        }),
+      }),
+      this.invalidateCache.invalidateQuery({
+        key: buildMessageCountKey().invalidate({
+          subscriberId,
+          _environmentId,
+        }),
+      }),
+    ]);
+
+    let deletedSubscriberCount = 0;
 
-      await this.subscriberRepository.delete({
-        _environmentId: subscriber._environmentId,
-        _organizationId: subscriber._organizationId,
-        subscriberId: subscriber.subscriberId,
+    await this.subscriberRepository.withTransaction(async () => {
+      /*
+       * Note about parallelism in transactions
+       *
+       * Running operations in parallel is not supported during a transaction.
+       * The use of Promise.all, Promise.allSettled, Promise.race, etc. to parallelize operations
+       * inside a transaction is undefined behaviour and should be avoided.
+       *
+       * Refer to https://mongoosejs.com/docs/transactions.html#note-about-parallelism-in-transactions
+       */
+      const { deletedCount } = await this.subscriberRepository.delete({
+        subscriberId,
+        _environmentId,
       });
 
+      deletedSubscriberCount = deletedCount;
+      if (deletedSubscriberCount === 0) {
+        return;
+      }
+
       await this.topicSubscribersRepository.delete({
-        _environmentId: subscriber._environmentId,
-        _organizationId: subscriber._organizationId,
-        externalSubscriberId: subscriber.subscriberId,
+        _environmentId,
+        externalSubscriberId: subscriberId,
       });
-    } catch (e) {
-      if (e instanceof DalException) {
-        throw new ApiException(e.message);
-      }
-      throw e;
+      await this.subscriberPreferenceRepository.delete({
+        _environmentId,
+        _subscriberId: subscriberId,
+      });
+      await this.preferenceRepository.delete({
+        _environmentId,
+        _subscriberId: subscriberId,
```

Comment:
in both cases you are not providing the right id
```suggestion
        _subscriberId: _subscriberId,
      });
      await this.preferenceRepository.delete({
        _environmentId,
        _subscriberId: _subscriberId,
```

Reply:
Good catch. We resolved it offline.